CONTRIBUTING.md

```diff
@@ -122,6 +122,48 @@ branch. Steps:
    git push -u origin my-feature
    ```
 
+## Local Developer Setup
+1. Make sure you have a compatible version of Python 3 installed
+2. From the project root, create a virtual environment and install all development dependencies
+
+   **Bash**
+    ```bash
+   # Create virtual environment
+   python3 -m venv venv
+   
+   # Activate the virtual environment
+   source venv/bin/activate
+   
+   # Install development dependencies
+   pip install -U pip setuptools
+   pip install -r requirements.txt
+   ```
+   **Windows (CMD)**
+    ```cmd
+   rem Create virtual environment
+   python3 -m venv venv
```

Comment:
oh wait this is wrong, isn't it?  shouldn't it be `py -3.<version>`?

Reply:
Added more comments to handle the python vs py launcher situation.